AppInspector.RulesEngine/RulesVerifier.cs

```diff
@@ -148,12 +148,27 @@ public RuleStatus CheckIntegrity(ConvertedOatRule convertedOatRule)
                     {
                         throw new ArgumentException();
                     }
-
+                    
                     RegexOptions regexOpts =
-                        Utils.RegexModifierToRegexOptions(searchPattern.Modifiers ?? Array.Empty<string>());
+                        Utils.RegexModifierToRegexOptions(searchPattern.Modifiers);
+                    
+#if NET7_0_OR_GREATER
+                    if (_options.EnableNonBacktrackingRegex)
+                    {
+                        regexOpts |= RegexOptions.NonBacktracking;
+                    }
+#endif
 
                     _ = new Regex(searchPattern.Pattern, regexOpts);
                 }
+                catch (NotSupportedException e)
+                {
+                    // Regex requires backtracking, try without non-backtracking
+                    _logger?.LogWarning(MsgHelp.GetString(MsgHelp.ID.VERIFY_RULES_REGEX_FAIL), rule.Id ?? "", searchPattern.Pattern ?? "", e.Message);
+                    errors.Add(MsgHelp.FormatString(MsgHelp.ID.VERIFY_RULES_REGEX_FAIL, rule.Id ?? "", searchPattern.Pattern ?? "", e.Message));
```

Comment:
Can't add this message to the errors. Any messages in the errors indicate validation failure. The Validation object doesn't currently have a field for informative messages like warnings, Errors are always assumed to be a validation failure.

Reply:
Actually this might not be true. Double checking the `Verified` field of the `RuleStatus` object it only checks the `OatIssues` field not the errors field.

Reply:
https://github.com/microsoft/ApplicationInspector/blob/c187a93438406c47be469285d731fca950d6d8df/AppInspector.RulesEngine/RuleStatus.cs#L11

Reply:
Just log then.